### ISSUE TITLE
phpspec-to-phpunit set not doing anything

### ISSUE THREAD

**Issue by X-Coder264** [rectorphp/rector#3931]
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | v0.7.63              |
| Installed as   | composer dependency                      |

I've tried to move the legacy part of my test suite which is written with phpspec to PHPUnit, but unfortunately the `phpspec-to-phpunit` set does not change the files at all (it only renames the file).

## Minimal PHP Code Causing Issue

<!-- Try to reproduce the issue using https://getrector.org/demo/ and post the URL here. -->
I tried to reproduce the issue using https://getrector.org/demo/ but it keeps saying ` Class "PhpSpec\ObjectBehavior" is missing. Complete it to code input, e.g. "class PhpSpec\ObjectBehavior {}" ` so I'm pasting the test here.

```php
<?php

namespace phpspec\JsonApiBundle\Paging;

use JsonApiBundle\Paging\PageLinksGenerator;
use JsonApiBundle\Paging\Paginator;
use Neomerx\JsonApi\Document\Link;
use PhpSpec\ObjectBehavior;
use Symfony\Bundle\FrameworkBundle\Routing\Router;
use Symfony\Component\Routing\Generator\UrlGeneratorInterface;

/**
 * Class PageLinksGeneratorSpec
 */
class PageLinksGeneratorSpec extends ObjectBehavior
{
    public function it_is_initializable()
    {
        $this->shouldHaveType(PageLinksGenerator::class);
    }

    public function let(Router $router)
    {
        $this->beConstructedWith($router);
    }

    public function it_should_return_only_self_link_when_no_results(Router $router)
    {
        $paginator = new Paginator(
            10,
            0,
            1,
            0,
            'route_name'
        );

        $router->generate($paginator->getRouteName(), [], UrlGeneratorInterface::ABSOLUTE_URL)
            ->shouldBeCalled()->willReturn('http://api.foo.loc/bar/1.0.0/');

        $expectations = [
            'self' => new Link('http://api.foo.loc/bar/1.0.0/'),
            'first' => null,
            'prev' => null,
            'next' => null,
            'last' => null,
        ];

        $this->generateLinks($paginator)->shouldBeLike($expectations);
    }
}
```

## Expected Behaviour

I was expecting Rector to give me a PHPUnit test but the only thing it does is to rename the file `PageLinksGeneratorSpec.php` to `PageLinksGeneratorTest.php`. The content of the file remained the same.

Output from Rector:

```
vendor/bin/rector process tests/Legacy/phpspec/JsonApiBundle/ --set phpspec-to-phpunit --debug
File "/webroot_data/webroot/vendor/autoload.php" is about to be loaded in "AutoloadIncluder::includeCwdVendorAutoloadIfExists()" on line 102"
File "/webroot_data/webroot/vendor/rector/rector/../../autoload.php" is about to be loaded in "AutoloadIncluder::autoloadProjectAutoloaderFile()" on line 136"
File "/webroot_data/webroot/vendor/autoload.php" is about to be loaded in "AutoloadIncluder::includeCwdVendorAutoloadIfExists()" on line 102"
File "/webroot_data/webroot/vendor/rector/rector/../../autoload.php" is about to be loaded in "AutoloadIncluder::autoloadProjectAutoloaderFile()" on line 136"
Rector v0.7.63

[parsing] tests/Legacy/phpspec/JsonApiBundle/Paging/PageLinksGeneratorSpec.php
[refactoring] tests/Legacy/phpspec/JsonApiBundle/Paging/PageLinksGeneratorSpec.php
    [applying] Rector\PhpSpecToPHPUnit\Rector\Class_\PhpSpecClassToPHPUnitClassRector
    [applying] Rector\PhpSpecToPHPUnit\Rector\Class_\AddMockPropertiesRector
    [applying] Rector\PhpSpecToPHPUnit\Rector\MethodCall\PhpSpecMocksToPHPUnitMocksRector
    [applying] Rector\PhpSpecToPHPUnit\Rector\ClassMethod\PhpSpecMethodToPHPUnitMethodRector
    [applying] Rector\PhpSpecToPHPUnit\Rector\MethodCall\PhpSpecPromisesToPHPUnitAssertRector
    [applying] Rector\PhpSpecToPHPUnit\Rector\ClassMethod\MockVariableToPropertyFetchRector
PHP Deprecated:  Function ReflectionType::__toString() is deprecated in /webroot_data/webroot/vendor/rector/rector/rules/php-spec-to-phpunit/src/PHPUnitTypeDeclarationDecorator.php on line 33
Deprecated: Function ReflectionType::__toString() is deprecated in /webroot_data/webroot/vendor/rector/rector/rules/php-spec-to-phpunit/src/PHPUnitTypeDeclarationDecorator.php on line 33
[refactoring with file system] tests/Legacy/phpspec/JsonApiBundle/Paging/PageLinksGeneratorSpec.php
[post rectors] tests/Legacy/phpspec/JsonApiBundle/Paging/PageLinksGeneratorSpec.php
[printing] tests/Legacy/phpspec/JsonApiBundle/Paging/PageLinksGeneratorSpec.php


                                                                                                                        
 [WARNING] File "tests/Legacy/phpspec/JsonApiBundle/Paging/PageLinksGeneratorSpec.php" was removed                      
                                                                                                                        

 ! [NOTE] 1 files were added/removed                                                                                    

                                                                                                                        
 [OK] Rector is done! 1 file has been changed. 
```

**Comment by dobryy** [comment]
@TomasVotruba I added fixture for this issue and the test is passing (#3937). Looks like some configuration/environment issue.

**Comment by X-Coder264** [comment]
I'm not familiar with the internals of Rector so some tips where to look in order to debug this would be greatly appreciated. I'm currently going through the entire flow (including vendor code like nikic's PHP parser) with Xdebug but without some knowledge of the internals it's hard to pinpoint where things start to go wrong.

**Comment by TomasVotruba** [comment]
No need to debug, the output you shared is good enough:

```
PHP Deprecated:  Function ReflectionType::__toString() is deprecated in /webroot_data/webroot/vendor/rector/rector/rules/php-spec-to-phpunit/src/PHPUnitTypeDeclarationDecorator.php on line 33
Deprecated: Function ReflectionType::__toString() is deprecated in /webroot_data/webroot/vendor/rector/rector/rules/php-spec-to-phpunit/src/PHPUnitTypeDeclarationDecorator.php on line 33
```  

So that line need to be fixed to be compatible with PHP 7.2-8.0.

**Comment by TomasVotruba** [comment]
Closing as no feedback for last 3 months
